tests/test_config.py

```diff
@@ -1868,6 +1868,46 @@ def test_quick_search_model_specific(self):
         with self.assertRaises(TritonModelAnalyzerException):
             self._evaluate_config(args, yaml_content, subcommand='profile')
 
+    def test_model_weightings(self):
+        """
+        Test that weightings can be specified only per model in the YAML
+        """
+        args = [
+            'model-analzyer', 'profile', '--model-repository', 'cli-repository',
+            '--run-config-search-mode', 'quick', '-f', 'path-to-config-file'
+        ]
+
+        # Global weighting - illegal
+        yaml_content = """
+        weighting: 3
+
+        profile_models:
+          resnet50_libtorch:
+            objectives:
+              perf_latency_p99: 1
+          add_sub:
+            objectives:
+              perf_throughput: 1
+        """
+
+        with self.assertRaises(TritonModelAnalyzerException):
+            self._evaluate_config(args, yaml_content, subcommand='profile')
+
+        # Model specific weighting - legal
+        yaml_content = """
+        profile_models:
+          resnet50_libtorch:
+            weighting: 3
```

Comment:
This looks good and makes a lot more sense now.  What is missing is documentation. Can you add a subsection to the multi-model page about weighting with a simple example, and also document the weighting option (maybe in config.md? I'm not sure where to put something that can only be specified per-model)

Reply:
Yes, documentation was going to be my next step. Just wasn't sure if it should be part of this story or a follow on. I can make it part of this story if that makes the most sense.

Reply:
Please review the added documenation.